### PR TITLE
pipeline run cleaner removes only finished pipeline runs, which

### PR DESCRIPTION
**Pull request by rcerven**
completed certain time ago

* CLOUDBLD-8142

Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

**Review by lkolacek — APPROVED** [review]

File: templates/pipeline_run_pruner.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+set -exuo pipefail
+
+delete_threshold=$(date --date="{{ osbs_pruner_pipeline_runs_minutes_old }} minutes ago" --iso-8601=seconds --utc)
+OC='oc -n {{ osbs_ocp_namespace }}'
+
+old_pipelineruns=$(
+    $OC get pipelinerun -o json | \
+    jq '.items[] | select(.status.completionTime != null) | select(.status.completionTime < "'${delete_threshold}'") | .metadata.name' -r | \
```
Comment (chmeliik) [inline templates/pipeline_run_pruner.sh:8]: Comparing timestamps (https://stedolan.github.io/jq/manual/#Dates) might be more robust, but this should work too

File: templates/pipeline_run_pruner.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+then
+    echo "Nothing to remove."
+    exit 0
+else
+    echo -e "${old_pipelineruns}"
+fi
+
+count=$(wc -l <<<${old_pipelineruns})
+echo "SUMMARY: ${count} Pipeline runs will be removed"
+echo ${old_pipelineruns} | xargs -n 10 $OC delete pipelinerun
```
Comment (chmeliik) [inline templates/pipeline_run_pruner.sh:22]: Does this also delete the associated TaskRuns etc?

Comment (rcerven) [reply templates/pipeline_run_pruner.sh:22]: yes removing pipeline run removes all related runs/pod/mounts

File: templates/pipeline_run_pruner.sh (hunk 1 above)
Comment (rcerven) [reply templates/pipeline_run_pruner.sh:8]: we are using it like this all the time and it works

File: templates/pipeline_run_pruner.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+set -exuo pipefail
+
+delete_threshold=$(date --date="{{ osbs_pruner_pipeline_runs_minutes_old }} minutes ago" --iso-8601=seconds --utc)
```
[the remark this answers is not in the document]
Comment (rcerven) [reply templates/pipeline_run_pruner.sh:3]: updated now

File: templates/pipeline_run_pruner.sh (hunk 1 above)
Comment (chmeliik) [reply templates/pipeline_run_pruner.sh:8]: Ack, works for me

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Pull request merged**